File: PracticasPDOORuby/Practica1_Ruby/lib/Irrgarten/shield.rb
# frozen_string_literal: true
module Irrgarten
  class Shield
    def initialize(protection, uses)
      @protection = protection.to_f
      @uses = uses.to_i
    end

    def protect()
      return 0.0 if @uses <= 0
      @uses -= 1
      @protection
    end

    def to_s
      return "S[" + @protection.to_s + ", " + @uses.to_s + "]"
    end

    def discard
      Dice.discard_element(@uses)
    end

    protected
    attr_reader :protection, :uses
  end
end

File: PracticasPDOOJava/Practica4_Java/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author Joaquín Cruz Lorenzo
 */
public class Shield {
    // Atributos de instancia privados
    private float protection;
    private int uses;
    
    // Constructor
    public Shield(float protection, int uses) {
        this.protection = protection;
        this.uses = uses;
    }
    
    public float protect(){
        if (uses > 0){
            uses--;
            return protection;
        }
        
        return 0.0f;
    }

    @Override
    public String toString() {
        return "S[" + "protection=" + protection + ", uses=" + uses + ']';
    }
    
    public boolean discard() {
        return Dice.discardElement(uses);
    }
}
